edx_exams/apps/core/api.py

```diff
@@ -158,6 +158,25 @@ def get_exam_attempt_time_remaining(exam_attempt, now=None):
     return secs_to_end
 
 
+def check_if_exam_timed_out(exam_attempt):
+    """
+    If an exam attempt's time remaining = 0 AND is in-progress/not submitted,
+    change the status of an exam attempt to 'submitted', then return attempt_id.
+    Return None otherwise.
+    """
+    IN_PROGRESS_STATUSES = [
+        ExamAttemptStatus.started,
+        ExamAttemptStatus.ready_to_submit,
+    ]
+    if exam_attempt.status in IN_PROGRESS_STATUSES and get_exam_attempt_time_remaining(exam_attempt) == 0:
```

Comment:
Question: I believe that get_exam_attempt_time_remaining will return 0 even if the time remaining has past. I am guessing that that is okay/expected?